models/demos/t3000/falcon40b/tt/falcon_decoder.py

```diff
@@ -32,17 +33,17 @@ def __init__(
         self.hidden_size = config.hidden_size
         self.state_dict = state_dict
         self.base_url = base_url
-        self.devices = devices
+        self.device_mesh = device_mesh
         self.layer_num = layer_num
         self.max_position_embeddings = max_position_embeddings
         self.model_config = model_config
-        self.num_devices = len(devices)
+        self.num_devices = len(device_mesh.get_device_ids())
```

Comment:
I think we don't need num_devices anymore, right? please check here and also in all other modules where we had it before, remove if we don't use it